GliaWidgets/SecureConversations/SecureConversations.WelcomeView.swift

```diff
@@ -3,30 +3,483 @@ import UIKit
+                titleLabel.trailingAnchor.constraint(
+                    lessThanOrEqualTo: safeAreaLayoutGuide.trailingAnchor,
+                    constant: -Self.sideMargin
+                )
+            ])
+            rootStackView.setCustomSpacing(16, after: titleLabel)
+
+        }
+
+        func defineSubtitleLabelLayout() {
+            rootStackView.addArrangedSubview(subtitleLabel)
+            NSLayoutConstraint.activate([
+                subtitleLabel.trailingAnchor.constraint(
+                    lessThanOrEqualTo: safeAreaLayoutGuide.trailingAnchor,
+                    constant: -Self.sideMargin
+                )
+            ])
+            rootStackView.setCustomSpacing(16, after: subtitleLabel)
+
+        }
+
+        func defineCheckMessagesButtonLayout() {
+            rootStackView.addArrangedSubview(checkMessagesButton)
+            rootStackView.setCustomSpacing(56, after: checkMessagesButton)
+        }
+
+        func defineMessageTitleStackViewLayout() {
+            rootStackView.addArrangedSubview(messageTitleStackView)
+            rootStackView.setCustomSpacing(5, after: messageTitleStackView)
+        }
+
+        func defineMessageTitleLabelLayout() {
+            NSLayoutConstraint.activate([
+                messageTitleLabel.trailingAnchor.constraint(
+                    equalTo: safeAreaLayoutGuide.trailingAnchor,
+                    constant: -Self.filePickerButtonSize - Self.sideMargin
+                )
+            ])
+        }
+
+        func defineFilePickerButtonLayout() {
+            NSLayoutConstraint.activate([
+                filePickerButton.widthAnchor.constraint(equalToConstant: Self.filePickerButtonSize),
+                filePickerButton.heightAnchor.constraint(equalToConstant: Self.filePickerButtonSize)
+            ])
+        }
+
+        func defineMessageTextViewLayout() {
+            rootStackView.addArrangedSubview(messageTextView)
+            // In design space between bottom part of border
+            // and bottom part of screen is 170 and screen height 812.
+            // So since height is different for every other
+            // device, we should rely on ratio.
+            let bottomSpaceToScreenHeightRatio = 0.209
+            NSLayoutConstraint.activate([
+                messageTextView.trailingAnchor.constraint(
+                    equalTo: safeAreaLayoutGuide.trailingAnchor,
+                    constant: -Self.sideMargin
+                ),
+                messageTextView.heightAnchor.constraint(
+                    greaterThanOrEqualTo: heightAnchor,
+                    multiplier: bottomSpaceToScreenHeightRatio
+                )
+            ])
+            rootStackView.setCustomSpacing(16, after: messageTextView)
+
+        }
+
+        func defineSendMessageButtonLayout() {
+            rootStackView.addArrangedSubview(sendMessageButton)
+            let widthConstraint = sendMessageButton.widthAnchor.constraint(
```

Comment:
Since this is vertical UIStackView all subviews will be resized automatically to width, so I think you can drop this constraint at all. Let's try to do it?

Reply:
@dukhovnyi  `rootStackView.alignment` is `.leading`. That turns off automatic resizing of subviews. Having it as `.fill` leads to some breaking of constraints, especially during the change of screen orientation. So I would leave it as it is for now.